api/apps/api/src/modules/access-control/projects-acl/dto/update-user-role-project.dto.ts

```diff
@@ -0,0 +1,14 @@
+import { Roles } from '@marxan-api/modules/users/role.api.entity';
+import { IsEnum, IsString, IsUUID } from 'class-validator';
+
+export class UpdateUserRoleinProject {
+  @IsUUID()
+  userId!: string;
+
+  @IsString()
+  @IsEnum(Object.values(Roles))
+  roleName?:
+    | Roles.project_viewer
+    | Roles.project_contributor
+    | Roles.project_owner;
+}
```

Comment:
1 Shouldn't it include the `projectId` we are changing role for?
2 also, `role?: Roles` should be enough unless we want to restrict which roles are possible